### DISCUSSION TITLE
how to build a query like "Select * From Table Where ( ... ) OR ( ... )"

### DISCUSSION THREAD

**Discussion by abenbachir** [maxtoroq/DbExtensions#20] [Q&A]
does this framework suppport this kind of query :

SELECT \* 
FROM Orders  
WHERE  
(
    CustomerID = 'VINET' AND OrderDate < '2005/01/01 00:00:00'
)  
OR 
(
    (CustomerID = 'TOMSP')  AND 
    (OrderDate < '2004/06/30 00:00:00' OR  OrderDate > '2006/01/01 00:00:00')
)

**Comment by maxtoroq** [comment]
There's no support for breaking complex boolean expressions into method calls, so you need to write it as a single WHERE call:

``` csharp
var query = SQL
   .SELECT("*")
   .FROM("Orders")
   .WHERE(@"
(CustomerID = {0} AND OrderDate < {1})
OR 
(
(CustomerID = {2}) AND 
(OrderDate < {3} OR OrderDate > {4})
)", /* insert parameters here */)
```
